docs/natspec-format.rst

```diff
@@ -0,0 +1,216 @@
+
+.. _header-output:
+
+Documentation Output
+====================
+
+When parsed by the compiler, documentation such as the one from the
+above example will produce 2 different JSON files. One is meant to be
+consumed by the end user as a notice when a function is executed and the
+other to be used by the developer.
+
+Let us see a more full contract example.
+
+.. code::
+
+   /// @title This is the contract title.
+   /// @author Homer Simpson
+   contract GavCoin
+   {
+     /// @notice Send `(valueInmGAV / 1000).fixed(0,3)` GAV from the account of
+     /// `message.caller.address()`, to an account accessible only by `to.address()
+     /// @dev This should be the documentation of the function for the developer docs
+     /// @param to The address of the recipient of the GavCoin
+     /// @param valueInmGav The GavCoin value to send
+     function send(address to, uint256 valueInmGAV) {
+       if (balances[msg.sender] >= valueInmGAV) {
+         balances[to] += valueInmGAV;
+         balances[msg.sender] -= valueInmGAV;
+       }
+     }
+
+     /// @notice `(balanceInmGAV / 1000).fixed(0,3)` GAV is the total funds available to `who.address()`.
+     /// @param who The address of the person whose balance we check
+     /// @return The balance of the user provided as argument
+     function balance(address who) constant returns (uint256 balanceInmGAV) {
+       balanceInmGAV = balances[who];
+     }
+
+     mapping (address => uint256) balances;
+   }
+
+.. _header-user-doc:
+
+User Documentation
+------------------
+
+The above documentation will produce the following user documentation
+JSON file as output:
+
+.. code::
+
+   {
+     "source": "...",
+     "language": "Solidity",
+     "languageVersion": 1,
+     "methods": {
+       "send(address,uint256)": { "notice": "Send `(valueInmGAV / 1000).fixed(0,3)` GAV from the account of `message.caller.address()`, to an account accessible only by `to.address()`." },
+       "balance(address)": { "notice": "`(balanceInmGAV / 1000).fixed(0,3)` GAV is the total funds available to `who.address()`." }
+     },
+     "invariants": [
+       { "notice": "The sum total amount of GAV in the system is 1 million." }
+     ],
+     "construction": [
+       { "notice": "Endows `message.caller.address()` with 1m GAV." }
+     ]
+   }
+
+Note that the key by which to find the methods is the function's
+canonical signature as defined in the `Contract
+ABI <Ethereum-Contract-ABI#signature>`__ and not simply the function's
+name.
+
+.. _header-developer-doc:
+
+Developer Documentation
+-----------------------
+
+Apart from the user documentation file, a developer documentation JSON
+file should also be produced and should look like this:
+
+.. code::
+
+   {
+     "author": "Larry A. Gardner",
+     "title": "A simulator for trees",
+     "methods": {
+       "age(uint256)": {
+         "details": "The Alexandr N. Tetearing algorithm could increase precision"
+       }
+     },
+     "invariants": [
```

Comment:
Is such a concept present somewhere?

Reply:
About the example: Yes, I really like it, too!

Reply:
Ah, no the invariant is cruft. I need to rerun the program to get the correct output. Patch coming...

Reply:
Updated in 5c0eabe